back/app/jobs/request_confirmation_code_job.rb

```diff
@@ -10,10 +10,11 @@ def run(user, new_email: nil)
     raise 'User confirmation is disabled.' if !AppConfiguration.instance.feature_activated?('user_confirmation')
 
     LogActivityJob.perform_later(user, 'requested_confirmation_code', user, Time.now.to_i, payload: { new_email: new_email })
+
     if new_email
       user.new_email = new_email
-      user.email_confirmation_code_reset_count = 0
```

Comment:
This was the bug related to the `new_email` param mentioned in the description

Reply:
I don't think so. Imagine I am trying to confirm the email `adrien@email.com` without having access to it. I can just request a code for `adrien@email.com`, then request another one for `fake@email.com`, which resets the count, and then request another one for `adrien@email.com` as often as I want